### ISSUE TITLE
Come up with some guidelines on what should and shouldn't be placed in IRAM

### ISSUE THREAD

**Issue by jessebraham** [esp-rs/esp-hal#960]
Quoting @bugadani in #943:

> I want to state here, too, that I think singling out SPI here isn't very elegant. Ideally all code should be as fast as possible, these getting the `#[ram]` treatment seems very arbitrary (to some who doesn't know the context). IMO before we start moving stuff into instruction ram somewhat haphazardly, we should probably come up with a systematic approach to tackle the problem.

**Comment by jessebraham** [comment]
Erm, duplicate of #946 (oops 😁 )
